src/engine/enginebuffer.h

```diff
@@ -343,7 +343,8 @@ class EngineBuffer : public EngineObject {
     // This ControlProxys is created as parent to this and deleted by
     // the Qt object tree. This helps that they are deleted by the creating
     // thread, which is required to avoid segfaults.
-    ControlProxy* m_pPassthroughEnabled;
+    ControlProxy* m_pPassthroughEnabledControl;
+    bool m_passthroughEnabled;
```

Comment:
Why store and manage a redundant state instead of simply accessing the current state through `m_pPassthroughEnabledControl->toBool()`?

Reply:
IIRC reading a bool member is faster than querying a control on every call?

Reply:
If this would be necessary then `ControlProxy` should instead store a local copy of the value, but not all using code again and again.

I never understood how all the `ControlThing`s are intended to be used safely and correctly.

Reply:
There is almost no overhead on a 64 bit system, except the function call. On a 32 bit system a ring buffer is used, which causes one indirection when reading and some ring maimtannance when writing.

Reply:
No overhead, so I drop the bool.